### PR TITLE
Improve output from RegionOps.computeAllXYAreas

### PR DESCRIPTION
**Pull request by EarlinLutz**
The function
`constructAllXYRegionLoops(curvesAndRegions: AnyCurve | AnyCurve[]): SignedLoops[]`
take in unstructured curves, finds intersections among them, and returns xy area loops.

Its output now includes (in addition to the loops) an array identifying adjacency among the loops.

### CODE REVIEW

File: core/geometry/src/test/curve/RegionBoolean.test.ts [hunk 1]
```diff
@@ -242,35 +250,274 @@ describe("RegionBoolean", () => {
     GeometryCoreTestIO.saveGeometry(allGeometry, "Solids", "ParityRegionWithBadBoundary");
     expect(ck.getNumErrors()).equals(0);
   });
-});
-/**
+  it.only("SectioningExample", () => {
```
Comment (pmconne) [inline core/geometry/src/test/curve/RegionBoolean.test.ts:253]: Remove .only

File: core/geometry/src/test/curve/RegionBoolean.test.ts [hunk 2]
```diff
@@ -242,35 +250,274 @@ describe("RegionBoolean", () => {
...
+        GeometryCoreTestIO.captureCloneGeometry(allGeometry, closedAreas, x0, y0);
+        const regionsB = RegionOps.constructAllXYRegionLoops((closedAreas as unknown) as AnyRegion[]);
+        saveShiftedLoops(allGeometry, regionsB, x0, y0 + dy, dy, positiveLoopOffset, false);
+        x0 = 0;
+        y0 += 100.0;
+      }
+    GeometryCoreTestIO.saveGeometry(allGeometry, "RegionBoolean", "SectioningExampleFile");
+      expect(ck.getNumErrors()).equals(0);
+    });
+    it.only("SectioningLineStringApproach", () => {
```
Comment (pmconne) [inline core/geometry/src/test/curve/RegionBoolean.test.ts:336]: .only

File: core/geometry/src/test/curve/RegionBoolean.test.ts [hunk 3]
```diff
@@ -242,35 +250,274 @@ describe("RegionBoolean", () => {
...
+        x0 += 100.0;
+      }
+    GeometryCoreTestIO.saveGeometry(allGeometry, "RegionBoolean", "SectioningLineStringApproach");
+      expect(ck.getNumErrors()).equals(0);
+      });
+      //  in the input directories ...
+      //    Madhav.json is the whole input set for 5 sections.
+      //    sectionA..sectionE are splits of the separate sections.
+      //  Note that the corresponding .imjs files are reduced to linestrings and are accessed in other tests.
+      it.skip("SectioningJson", () => {
```
Comment (pmconne) [inline core/geometry/src/test/curve/RegionBoolean.test.ts:411]: Intentionally skipped?

**Review by pmconne — CHANGES_REQUESTED** [review]

**Review by bbastings — APPROVED** [review]

File: core/geometry/src/test/curve/RegionBoolean.test.ts (hunk 2 above)
Comment (EarlinLutz) [reply core/geometry/src/test/curve/RegionBoolean.test.ts]: (fixed)

File: core/geometry/src/test/curve/RegionBoolean.test.ts (hunk 3 above)
Comment (EarlinLutz) [reply core/geometry/src/test/curve/RegionBoolean.test.ts:412]: yes.  At the moment its purpose is part of helping the AI road section recognition developers figure out what's in their data.  It's console output is useful in nudging that forward but is not testing anything in the library.   As we understand more of what their fixup needs are that might evolve into tests of the ChainCollector.

File: core/geometry/src/test/curve/RegionBoolean.test.ts (hunk 1 above)
Comment (EarlinLutz) [reply core/geometry/src/test/curve/RegionBoolean.test.ts]: (fixed)

**Pull request merged**